lib/KubeResourceMeta.js

```diff
@@ -69,74 +70,79 @@ module.exports = class KubeResourceMeta {
   get kubeApiConfig() {
     return this._kubeApiConfig;
   }
+  get extraHeaders() {
+    return this._extraHeaders;
+  }
 
   hasVerb(verb) {
     return this.verbs.some(v => verb == v);
   }
+  addHeader(key, value) {
+    objectPath.set(this._extraHeaders, ['headers', key], value);
+  }
+  removeHeader(key) {
+    objectPath.del(this._extraHeaders, ['headers', key]);
+    if (Object.keys(objectPath.get(this._extraHeaders, 'headers', [])).length === 0) {
+      objectPath.del(this._extraHeaders, 'headers');
+    }
+  }
 
 
   async request(reqOpt) {
     this._logger.debug(`Request ${reqOpt.method || 'GET'} ${reqOpt.uri || reqOpt.url}`);
-    return request(merge(this._kubeApiConfig, reqOpt));
+    return request(merge.all([this._kubeApiConfig, reqOpt, this._extraHeaders]));
```

Comment:
```suggestion
    return request(merge.all([this._kubeApiConfig, this._extraHeaders, reqOpt]));
```

Thinking maybe reqOpt should take precedence over the extra headers.

Reply:
i guess that would make sense since reqOpts is more granular.